bcdaworker/worker/alr.go

```diff
@@ -158,38 +222,40 @@ func (a *AlrWorker) ProcessAlrJob(
 
 	// Creating channels for go routine
 	// c is buffered b/c IO operation is slower than unmarshalling
-	c := make(chan data, 1000) // 1000 rows before blocking
+	c := make(chan *alr.AlrFhirBulk, 1000) // 1000 rows before blocking
 	result := make(chan error)
 
 	// A go routine that will streamed data to write to disk.
 	// Reason for a go routine is to not block when writing, since disk writing is
 	// generally slower than memory access. We are streaming to keep mem lower.
-	go goWriter(c, w, marshaller, result)
+	go goWriter(ctx, a, c, fileMap, marshaller, result, resources[:], id)
 
 	// Marshall into JSON and send it over the channel
 	for i := range alrModels {
-		patient, observations := alr.ToFHIR(&alrModels[i], alrModels[i].Timestamp)
-		c <- data{patient, observations}
+		fhirBulk := alr.ToFHIR(&alrModels[i]) // Removed timestamp, but can be added back here
```

Comment:
I'd consider adding it back, we include this with non-ALR resources

Reply:
I think so too. I'll do this in the other ticket that implements r4.